lib/chef/provisioning/aws_driver/driver.rb

```diff
@@ -603,8 +603,8 @@ def load_balancer_for(lb_spec)
 
     def instance_for(machine_spec)
       if machine_spec.reference
-        if machine_spec.reference['driver_url'] != driver_url
```

Comment:
The underlying issue in #189 is that `driver_url` is no longer stored in the reference by machine, because it is using the new API exposed in chef-provisioning 1.0